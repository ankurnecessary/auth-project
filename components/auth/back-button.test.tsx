import { render, screen } from '@testing-library/react';
import BackButton from '@/components/auth/back-button'; // Adjust path accordingly
import '@testing-library/jest-dom';

describe('BackButton component', () => {
  const label = 'Go Back';
  const href = '/previous-page';

  it('renders BackButton with label and href', () => {
    render(<BackButton label={label} href={href} />);

    // Check if the button contains the label
    const linkElement = screen.getByRole('link', { name: label });
    expect(linkElement).toBeInTheDocument();
    expect(linkElement).toHaveTextContent(label);
    expect(linkElement).toHaveAttribute('href', href);
  });

  it('renders the Button component with correct classes', () => {
    render(<BackButton label={label} href={href} />);

    // Check if Button has the correct classes
    const linkElement = screen.getByRole('link', { name: label });
    expect(linkElement).toHaveClass('w-full font-normal');
  });

  it('uses the asChild prop to render Link as a child of Button', () => {
    render(<BackButton label={label} href={href} />);

    // Ensure the button contains a link
    const linkElement = screen.getByRole('link', { name: label });
    expect(linkElement).toBeInTheDocument(); // Check if Link is wrapped in a Button
  });
});
