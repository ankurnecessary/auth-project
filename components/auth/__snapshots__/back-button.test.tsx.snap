// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BackButton component matches snapshot when all attributes are provided 1`] = `
<DocumentFragment>
  <a
    class="inline-flex items-center justify-center whitespace-nowrap transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:pointer-events-none disabled:opacity-50 text-primary underline-offset-4 hover:underline h-8 rounded-md px-3 text-xs w-full font-normal"
    href="/previous-page"
  >
    Go Back
  </a>
</DocumentFragment>
`;
